zerver/lib/push_notifications.py

```diff
@@ -340,10 +341,16 @@ def get_alert_from_message(message):
 
 def get_apns_payload(message):
     # type: (Message) -> Dict[str, Any]
+    content = message.content
+    content_truncated = (len(content) > 200)
+    if content_truncated:
+        content = content[:200] + "..."
     return {
-        'alert': get_alert_from_message(message),
+        'alert': {
+            'title': get_alert_from_message(message),
+            'body': content,
+        },
         # TODO: set badge count in a better way
-        'badge': 1,
```

Comment:
Hmm -- by my reading of [the docs](https://developer.apple.com/library/content/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/PayloadKeyReference.html) (thanks for the link!), this change means we leave the badge unchanged at whatever it might currently be. So if the user has gotten some notifications with the old code, it'll be perpetually 1, and if they haven't, it'll be 0 (unbadged).

I'm not sure that's actually an improvement. If we want to not set any badges for now, we should say `'badge': 0`. Alternatively, just leave it as is in this PR.

Either way, writing some appropriate logic to set the badge count correctly to the number of unread notifications would be a great next thing to do. :-)

Reply:
I removed it as sending `badge: 1` will always set the badge as 1 even if the user has received 3 notifications of message, which is more confusing for the user so I removed the badge to not show the badges! 

Working on the badge system would be a little complex for tracking the part when the user has connected to the server through the mobile & resetting the badge at that point (maybe not), but currently the mobile app needs some more essential work so I may work on this at a later point of time :)

Reply:
OK, but this version wouldn't have actually removed the badge! From those docs:

> badge | Number | Include this key when you want the system to modify the badge of your app icon.If this key is not included in the dictionary, the badge is not changed. To remove the badge, set the value of this key to 0.

Some users -- like me :smile: -- already have the badge set. So if we want to remove it, we should set the key here with a value of 0.

If a user has received 3 notifications, I think a badge count of 1 is actually better and less confusing than no badge, even though 3 would be better. But once they've read those messages, then the badge count of 1 that doesn't go away is a problem -- so I'd be happy to merge a PR that cleared the badges for now.

Reply:
Does the badge comes as 1 even after executing the app? 
AFAIR, I didn't use any code for removing the badge count (in the app) 

If I work on this, I should should reset the badge count whenever the user comes online right & set the badge count to 0 (After storing the badge count)